### ISSUE TITLE
Cannot fetch profiles from remote Mastodon instances with AUTHORIZED_FETCH: `Undefined index: outbox` 

### ISSUE THREAD

**Issue by inga-lovinde** [pixelfed/pixelfed#2713]
When I'm trying to search for any beach.city user (e.g. `@anne@beach.city`), I'm getting "An error occured, results could not be loaded. Please try again later." (I tried doing that both on my local instance with v0.10.10, and on pixelfed.de)

In laravel.log, there is a message:
```
production.ERROR: Undefined index: outbox {"userId":4,"exception":[object] (ErrorException(code: 0): Undefined index: outbox at 

[2021-04-04 01:27:20] production.ERROR: Undefined index: outbox {"userId"':4, exception: [object] (ErrorException(code: 0): Undefined index: outbox at /var/www/pixelfed/app/Util/ActivityPub/Helpers.php:470)
[stacktrace]
#0 /var/www/pixelfed/app/Util/ActivityPub/Helpers.php(470): Illuminate\\Foundation\\Bootstrap\\HandleExceptions->handleError (8, 'Undefined index...', '/var/www/pixelf...', 470, Array)
#1 /var/www/pixelfed/vendor/laravel/framework/src/Illuminate/Database/Concerns/ManagesTransactions.php(29): App\\Util\\ActivityPub\\Helpers::App\\Util\\ActivityPub\\{closure}(Object(Illuminate\\Database\\PostgresConnection))
#2 /var/www/pixelfed/vendor/laravel/framework/src/Illuminate/Database/DatabaseManager.php(371): Illuminate\\Database\\Connection->transaction(Object(Closure))
#3 /var/www/pixelfed/vendor/laravel/framework/src/Illuminate/Support/Facades/Facade.php(261): Illuminate\\Database\\DatabaseManager->__call('transaction', Array)
#4 /var/www/pixelfed/app/util/ActivityPub/Helpers.php(479): Illuminate\\Support\\Facades\\Facade::__callStatic('transaction', Array)
#5 /var/www/pixelfed/vendor/laravel/framework/src/Illuminate/Cache/Repository.php(385): App\\Util\\ActivityPub\\Helpers::App\\Util\\ActivityPub\\{closure}()
#6 /var/www/pixelfed/vendor/laravel/framework/src/Illuminate/Cache/CacheManager.php(392): Illuminate\\Cache\\Repository->remember('ap:profile:by_u...', Object(Illuminate\\Support\\Carbon), Object(Closure))
#7 /var/www/pixelfed/vendor/laravel/framework/src/Illuminate/Support/Facades/Facade.php(261): Illuminate\\Cache\\CacheManager->__call('remember’, Array)
#8 /var/www/pixelfed/app/Util/ActivityPub/Helpers.php(494): Illuminate\\Support\\Facades\\Facade::__callStatic('remember', Array)
#9 /var/www/pixelfed/app/Util/ActivityPub/Helpers.php(500): App\\Util\\ActivityPub\\Helpers::profileFirstOrNew('https: //beach.c...')
#10 /var/www/pixelfed/app/Services/WebfingerService.php(40): App\\Util\\ActivityPub\\Helpers::profileFetch('https://beach.c...')
#11 /var/www/pixelfed/app/Services/WebfingerService.php(20): App\\Services\\WebfingerService->run('@[REDACTED]@beach.city')
#12 /var/www/pixelfed/app/Http/Controllers/SearchController.php(285): App\\Services\\WebfingerService::lookup(‘@[REDACTED]@beach.city')
...
```

I also do not receive any interactions from these users, which might be related (but there seems to be no relevant error messages in laravel.log for these interactions).

Everything seems to work fine for other instances.

**Comment by inga-lovinde** [comment]
In Horizon, there is a following message for interactions from beach.city users:

```
App\Jobs\InboxPipeline\InboxValidator


ErrorException: Undefined index: outbox in /var/www/pixelfed/app/Util/ActivityPub/Helpers.php:470
--
Stack trace:
#0  /var/www/pixelfed/app/Util/ActivityPub/Helpers.php(470):  Illuminate\Foundation\Bootstrap\HandleExceptions->handleError(8,  'Undefined index...', '/var/www/pixelf...', 470, Array)
#1   /var/www/pixelfed/vendor/laravel/framework/src/Illuminate/Database/Concerns/ManagesTransactions.php(29):   App\Util\ActivityPub\Helpers::App\Util\ActivityPub\{closure}(Object(Illuminate\Database\PostgresConnection))
#2   /var/www/pixelfed/vendor/laravel/framework/src/Illuminate/Database/DatabaseManager.php(371):  Illuminate\Database\Connection->transaction(Object(Closure))
#3   /var/www/pixelfed/vendor/laravel/framework/src/Illuminate/Support/Facades/Facade.php(261):  Illuminate\Database\DatabaseManager->__call('transaction', Array)
#4  /var/www/pixelfed/app/Util/ActivityPub/Helpers.php(479):  Illuminate\Support\Facades\Facade::__callStatic('transaction', Array)
#5   /var/www/pixelfed/vendor/laravel/framework/src/Illuminate/Cache/Repository.php(385):  App\Util\ActivityPub\Helpers::App\Util\ActivityPub\{closure}()
#6   /var/www/pixelfed/vendor/laravel/framework/src/Illuminate/Cache/CacheManager.php(392):  Illuminate\Cache\Repository->remember('ap:profile:by_u...',  Object(Illuminate\Support\Carbon), Object(Closure))
#7   /var/www/pixelfed/vendor/laravel/framework/src/Illuminate/Support/Facades/Facade.php(261):  Illuminate\Cache\CacheManager->__call('remember', Array)
#8 /var/www/pixelfed/app/Util/ActivityPub/Helpers.php(494): Illuminate\Support\Facades\Facade::__callStatic('remember', Array)
#9  /var/www/pixelfed/app/Jobs/InboxPipeline/InboxValidator.php(170):  App\Util\ActivityPub\Helpers::profileFirstOrNew('https://beach.c...')
#10  /var/www/pixelfed/app/Jobs/InboxPipeline/InboxValidator.php(120):  App\Jobs\InboxPipeline\InboxValidator->verifySignature(Array,  Object(App\Profile), Array)
#11   /var/www/pixelfed/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(36):  App\Jobs\InboxPipeline\InboxValidator->handle()
#12   /var/www/pixelfed/vendor/laravel/framework/src/Illuminate/Container/Util.php(40):  Illuminate\Container\BoundMethod::Illuminate\Container\{closure}()
#13   /var/www/pixelfed/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(93):  Illuminate\Container\Util::unwrapIfClosure(Object(Closure))
#14   /var/www/pixelfed/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(37):   Illuminate\Container\BoundMethod::callBoundMethod(Object(Illuminate\Foundation\Application),  Array, Object(Closure))
#15   /var/www/pixelfed/vendor/laravel/framework/src/Illuminate/Container/Container.php(611):   Illuminate\Container\BoundMethod::call(Object(Illuminate\Foundation\Application),  Array, Array, NULL)
#16   /var/www/pixelfed/vendor/laravel/framework/src/Illuminate/Bus/Dispatcher.php(128):  Illuminate\Container\Container->call(Array)
#17   /var/www/pixelfed/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(128):   Illuminate\Bus\Dispatcher->Illuminate\Bus\{closure}(Object(App\Jobs\InboxPipeline\InboxValidator))
#18   /var/www/pixelfed/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(103):   Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(App\Jobs\InboxPipeline\InboxValidator))
#19   /var/www/pixelfed/vendor/laravel/framework/src/Illuminate/Bus/Dispatcher.php(132):  Illuminate\Pipeline\Pipeline->then(Object(Closure))
#20   /var/www/pixelfed/vendor/laravel/framework/src/Illuminate/Queue/CallQueuedHandler.php(118):   Illuminate\Bus\Dispatcher->dispatchNow(Object(App\Jobs\InboxPipeline\InboxValidator),  false)
#21   /var/www/pixelfed/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(128):   Illuminate\Queue\CallQueuedHandler->Illuminate\Queue\{closure}(Object(App\Jobs\InboxPipeline\InboxValidator))
#22   /var/www/pixelfed/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(103):   Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(App\Jobs\InboxPipeline\InboxValidator))
#23   /var/www/pixelfed/vendor/laravel/framework/src/Illuminate/Queue/CallQueuedHandler.php(120):  Illuminate\Pipeline\Pipeline->then(Object(Closure))
#24   /var/www/pixelfed/vendor/laravel/framework/src/Illuminate/Queue/CallQueuedHandler.php(70):   Illuminate\Queue\CallQueuedHandler->dispatchThroughMiddleware(Object(Illuminate\Queue\Jobs\RedisJob),  Object(App\Jobs\InboxPipeline\InboxValidator))
#25   /var/www/pixelfed/vendor/laravel/framework/src/Illuminate/Queue/Jobs/Job.php(98):   Illuminate\Queue\CallQueuedHandler->call(Object(Illuminate\Queue\Jobs\RedisJob),  Array)
#26 /var/www/pixelfed/vendor/laravel/framework/src/Illuminate/Queue/Worker.php(410): Illuminate\Queue\Jobs\Job->fire()
#27   /var/www/pixelfed/vendor/laravel/framework/src/Illuminate/Queue/Worker.php(360):  Illuminate\Queue\Worker->process('redis',  Object(Illuminate\Queue\Jobs\RedisJob),  Object(Illuminate\Queue\WorkerOptions))
#28   /var/www/pixelfed/vendor/laravel/framework/src/Illuminate/Queue/Worker.php(158):   Illuminate\Queue\Worker->runJob(Object(Illuminate\Queue\Jobs\RedisJob),  'redis', Object(Illuminate\Queue\WorkerOptions))
#29   /var/www/pixelfed/vendor/laravel/framework/src/Illuminate/Queue/Console/WorkCommand.php(117):  Illuminate\Queue\Worker->daemon('redis', 'high',  Object(Illuminate\Queue\WorkerOptions))
#30   /var/www/pixelfed/vendor/laravel/framework/src/Illuminate/Queue/Console/WorkCommand.php(101):  Illuminate\Queue\Console\WorkCommand->runWorker('redis', 'high')
#31 /var/www/pixelfed/vendor/laravel/horizon/src/Console/WorkCommand.php(51): Illuminate\Queue\Console\WorkCommand->handle()
#32   /var/www/pixelfed/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(36):  Laravel\Horizon\Console\WorkCommand->handle()
#33   /var/www/pixelfed/vendor/laravel/framework/src/Illuminate/Container/Util.php(40):  Illuminate\Container\BoundMethod::Illuminate\Container\{closure}()
#34   /var/www/pixelfed/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(93):  Illuminate\Container\Util::unwrapIfClosure(Object(Closure))
#35   /var/www/pixelfed/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(37):   Illuminate\Container\BoundMethod::callBoundMethod(Object(Illuminate\Foundation\Application),  Array, Object(Closure))
#36   /var/www/pixelfed/vendor/laravel/framework/src/Illuminate/Container/Container.php(611):   Illuminate\Container\BoundMethod::call(Object(Illuminate\Foundation\Application),  Array, Array, NULL)
#37   /var/www/pixelfed/vendor/laravel/framework/src/Illuminate/Console/Command.php(136):  Illuminate\Container\Container->call(Array)
#38  /var/www/pixelfed/vendor/symfony/console/Command/Command.php(256):  Illuminate\Console\Command->execute(Object(Symfony\Component\Console\Input\ArgvInput),  Object(Illuminate\Console\OutputStyle))
#39   /var/www/pixelfed/vendor/laravel/framework/src/Illuminate/Console/Command.php(121):   Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput),  Object(Illuminate\Console\OutputStyle))
#40  /var/www/pixelfed/vendor/symfony/console/Application.php(971):  Illuminate\Console\Command->run(Object(Symfony\Component\Console\Input\ArgvInput),  Object(Symfony\Component\Console\Output\ConsoleOutput))
#41  /var/www/pixelfed/vendor/symfony/console/Application.php(290):  Symfony\Component\Console\Application->doRunCommand(Object(Laravel\Horizon\Console\WorkCommand),  Object(Symfony\Component\Console\Input\ArgvInput),  Object(Symfony\Component\Console\Output\ConsoleOutput))
#42  /var/www/pixelfed/vendor/symfony/console/Application.php(166):  Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput),  Object(Symfony\Component\Console\Output\ConsoleOutput))
#43   /var/www/pixelfed/vendor/laravel/framework/src/Illuminate/Console/Application.php(92):   Symfony\Component\Console\Application->run(Object(Symfony\Component\Console\Input\ArgvInput),  Object(Symfony\Component\Console\Output\ConsoleOutput))
#44   /var/www/pixelfed/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(129):   Illuminate\Console\Application->run(Object(Symfony\Component\Console\Input\ArgvInput),  Object(Symfony\Component\Console\Output\ConsoleOutput))
#45  /var/www/pixelfed/artisan(37):  Illuminate\Foundation\Console\Kernel->handle(Object(Symfony\Component\Console\Input\ArgvInput),  Object(Symfony\Component\Console\Output\ConsoleOutput))
#46 {main}
```

Data is:
```
 {
"command": "O:37:"App\Jobs\InboxPipeline\InboxValidator":15:{s:11:"*username";s:11:"IngaLovinde";s:10:"*headers";a:9:{s:14:"content-length";a:1:{i:0;s:4:"2585";}s:10:"connection";a:1:{i:0;s:10:"Keep-Alive";}s:9:"signature";a:1:{i:0;s:482:"keyId="https://beach.city/users/owl#main-key",algorithm="rsa-sha256",headers="(request-target) host date digest content-type",signature="Bc3n7yIFoUOnRZ+op+nnInIGH4hvHZb5cyxlYCjTF/kGagxUIR1C/xLai3H0EI6+LdG/C54uaVKwfnyqqbprMaShh3YgGuDuxVdDZ0r6VIdeH0jPm29xb2SHNizYteWMCJkjguetAggBpU/kvxtdPp6RY2q0s4twhgyRP+YQZux6inWsnKJY4Ie9ouO7CQEmbZzNsFZAkr7PoljJFaNti5kZMfog8ScW1ab563ExNMLExa6BMafKrqaQas/xNynvC67y5C+5Iysd2/XAmiS8h2hlgWUBO4wv6QpWv5QBFlDmjGjSHX1BFmC0DH8J59d/dMzJZvrVkuJ91SBkCtqgyg=="";}s:12:"content-type";a:1:{i:0;s:25:"application/activity+json";}s:6:"digest";a:1:{i:0;s:52:"SHA-256=N4JAPpzEaNJq1sW0dvmzuRYjGrXK/8h/zUmWvNmp/os=";}s:15:"accept-encoding";a:1:{i:0;s:4:"gzip";}s:4:"date";a:1:{i:0;s:29:"Sun, 04 Apr 2021 09:40:14 GMT";}s:4:"host";a:1:{i:0;s:13:"quite.exposed";}s:10:"user-agent";a:1:{i:0;s:68:"http.rb/4.4.1 (Mastodon/3.2.0+beachcity1.20.0; +https://beach.city/)";}}s:10:"*payload";s:2585:"{"@context":["https://www.w3.org/ns/activitystreams",{"ostatus":"http://ostatus.org#","atomUri":"ostatus:atomUri","inReplyToAtomUri":"ostatus:inReplyToAtomUri","conversation":"ostatus:conversation","sensitive":"as:sensitive","toot":"http://joinmastodon.org/ns#","votersCount":"toot:votersCount"}],"id":"https://beach.city/users/owl/statuses/106006394594307592/activity","type":"Create","actor":"https://beach.city/users/owl","published":"2021-04-04T09:40:14Z","to":["https://beach.city/users/owl/followers"],"cc":["https://www.w3.org/ns/activitystreams#Public","https://quite.exposed/users/IngaLovinde","https://embracing.space/users/IngaLovinde"],"object":{"id":"https://beach.city/users/owl/statuses/106006394594307592","type":"Note","summary":null,"inReplyTo":"https://embracing.space/users/IngaLovinde/statuses/106004578181546382","published":"2021-04-04T09:40:14Z","url":"https://beach.city/@owl/106006394594307592","attributedTo":"https://beach.city/users/owl","to":["https://beach.city/users/owl/followers"],"cc":["https://www.w3.org/ns/activitystreams#Public","https://quite.exposed/users/IngaLovinde","https://embracing.space/users/IngaLovinde"],"sensitive":false,"atomUri":"https://beach.city/users/owl/statuses/106006394594307592","inReplyToAtomUri":"https://embracing.space/users/IngaLovinde/statuses/106004578181546382","conversation":"tag:beach.city,2021-04-04:objectId=15841779:objectType=Conversation","content":"<p><span class=\"h-card\"><a href=\"https://embracing.space/@IngaLovinde\" class=\"u-url mention\">@<span>IngaLovinde</span></a></span> <span class=\"h-card\"><a href=\"https://quite.exposed/IngaLovinde\" class=\"u-url mention\">@<span>IngaLovinde</span></a></span> oh no 😞</p>","contentMap":{"is":"<p><span class=\"h-card\"><a href=\"https://embracing.space/@IngaLovinde\" class=\"u-url mention\">@<span>IngaLovinde</span></a></span> <span class=\"h-card\"><a href=\"https://quite.exposed/IngaLovinde\" class=\"u-url mention\">@<span>IngaLovinde</span></a></span> oh no 😞</p>"},"attachment":[],"tag":[{"type":"Mention","href":"https://embracing.space/users/IngaLovinde","name":"@IngaLovinde@embracing.space"},{"type":"Mention","href":"https://quite.exposed/users/IngaLovinde","name":"@IngaLovinde@quite.exposed"}],"replies":{"id":"https://beach.city/users/owl/statuses/106006394594307592/replies","type":"Collection","first":{"type":"CollectionPage","next":"https://beach.city/users/owl/statuses/106006394594307592/replies?only_other_accounts=true&page=true","partOf":"https://beach.city/users/owl/statuses/106006394594307592/replies","items":[]}}}}";s:7:"timeout";i:60;s:5:"tries";i:1;s:3:"job";N;s:10:"connection";N;s:5:"queue";s:4:"high";s:15:"chainConnection";N;s:10:"chainQueue";N;s:19:"chainCatchCallbacks";N;s:5:"delay";N;s:11:"afterCommit";N;s:10:"middleware";a:0:{}s:7:"chained";a:0:{}}",
"commandName": "App\Jobs\InboxPipeline\InboxValidator"
}

**Comment by trwnh** [comment]
this appears to be because of AUTHORIZED_FETCH -- the response that pixelfed is getting doesn't contain an `outbox`, probably because it doesn't contain anything else either. likely what is happening is that pixelfed is sending an HTTP Signature which fails to validate at beach.city

**Comment by dansup** [comment]
@inga-lovinde Fixed in #4504
